contrib/aws/aws-sdk-go/aws/aws_test.go

```diff
@@ -142,3 +144,35 @@ func TestAnalyticsSettings(t *testing.T) {
 		assertRate(t, mt, 0.23, WithAnalyticsRate(0.23))
 	})
 }
+
+func TestRetries(t *testing.T) {
+	cfg := aws.NewConfig().
+		WithRegion("us-west-2").
+		WithDisableSSL(true).
+		WithCredentials(credentials.AnonymousCredentials)
+
+	session := WrapSession(session.Must(session.NewSession(cfg)))
+	expectedError := errors.New("an error")
+	session.Handlers.Send.PushBack(func(r *request.Request) {
+		r.Error = expectedError
+		r.Retryable = aws.Bool(true)
+	})
+
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	ctx := context.Background()
+	s3api := s3.New(session)
+	req, _ := s3api.GetObjectRequest(&s3.GetObjectInput{
+		Bucket: aws.String("BUCKET"),
+		Key:    aws.String("KEY"),
+	})
+	req.SetContext(ctx)
+	err := req.Send()
+
+	assert.Equal(t, 3, req.RetryCount)
+	assert.Same(t, expectedError, err)
+	assert.Len(t, mt.OpenSpans(), 0)
```

Comment:
Hold on a minute. Sorry about this. Let's rewind back. Why do we even need this method again?

Reply:
We need it because without it there's no way to write a test for the fix for #769. This module requires that any span that gets opened needs to be explicitly closed, otherwise the spans leak. The change to `mockTracer` adds reference  tracking.

I guess this could be changed to be something like `OpenSpanCount()` or `HasOpenSpans()`, but returning the full list of spans outstanding is the most flexible.

Reply:
No I think we're good like this. No need to change. Thanks for explaining. I was a bit out of context here.